src/mam4xx/gasaerexch_soaexch.hpp

```diff
@@ -61,6 +61,259 @@ void soa_equilib_mixing_ratio_no_solute(const Real &T_in_K,     // in
   g0_soa = pstd_in_Pa * p0_soa / p_in_Pa;
 }
 
+// NOTE: This is different from box model.
+constexpr int npoa =
+    1; //   number of differently tagged primary-organic   aerosol species
+constexpr int nsoa =
+    1; // number of differently tagged secondary-organic aerosol species
+// from: modal_aero_amicphys_control
+constexpr int max_mode = AeroConfig::num_modes() + 1;
+constexpr int max_gas = nsoa + 1;
+//  the +4 in max_aer are dst, ncl, so4, mom
+constexpr int nbc =
+    1; // number of differently tagged black-carbon      aerosol species
+constexpr int max_aer = nsoa + npoa + nbc + 4;
+constexpr int naer = nsoa + 1;
+constexpr int iaer_pom =
+    naer + 1 - 1; // -1 for fotran to c++ indexing conversion.
+constexpr int ntot_soamode = 4;
+
+KOKKOS_INLINE_FUNCTION
+void mam_soaexch_1subarea(const Real dtsubstep,                 // in
+                          const Real temp,                      // in
+                          const Real pmid,                      // in
+                          Real qgas_cur[max_gas],               // in/out
+                          Real qgas_avg[max_gas],               // in/out
+                          Real qaer_cur[max_aer][max_mode],     // in/out
+                          Real qnum_cur[max_mode],              // in/out
+                          Real qwtr_cur[max_mode],              // in/out
+                          const Real uptkaer[max_gas][max_mode] // in
+) {
+
+  constexpr int ntot_poaspec = npoa;
+  constexpr int ntot_soaspec = nsoa;
+  int niter_max = 1000;
+  int niter = 0;
+  const Real dtfull = dtsubstep;
+  Real tcur = 0.0;
+  Real dtcur = 0.0;
+  Real dtsum_qgas_avg = 0.0;
+  bool skip_soamode[max_mode] = {true};
+  Real qgas_prv[max_gas] = {0.0};
+  Real qaer_prv[max_aer][max_mode] = {{0.0}};
+  Real uptkaer_soag_tmp[nsoa][max_mode] = {{0.0}};
+  Real a_ooa_sum_tmp[max_mode] = {0.0};
+  Real a_opoa[max_mode] = {0.0};
+  Real a_soa[ntot_soaspec][max_mode] = {{0.0}};
+  Real a_soa_tmp[ntot_soaspec][max_mode] = {{0.0}};
+  Real beta[ntot_soaspec][max_mode] = {{0.0}};
+  Real delh_vap_soa[ntot_soaspec] = {156.0e3};
+  Real del_g_soa_tmp[ntot_soaspec] = {0.0};
+  Real g0_soa[ntot_soaspec] = {0.0};
+  Real g_soa[ntot_soaspec] = {0.0};
+  Real g_star[ntot_soaspec][max_mode] = {{0.0}};
+  Real opoa_frac[ntot_poaspec][max_mode] = {{0.1}};
+  Real phi[ntot_soaspec][max_mode] = {{0.0}};
+  Real p0_soa[ntot_soaspec] = {1.0e-10};
+  Real p0_soa_298[ntot_soaspec] = {1.0e-10};
+  Real sat[ntot_soaspec][max_mode] = {{0.0}};
```

Comment:
You probably already know, but I found out in the past that initializing arrays like this doesn't actually initialize all array elements. C++ initializes the first element to the value in the curly brackets and the rest are initialized to zero. It is not guaranteed to work on GPU.

Reply:
Thanks for catching this. It has been fixed in the code.